src/Assessment/Processor/ProcessDatasets.cs

```diff
@@ -1243,6 +1249,277 @@ private void UpdateIndividualWebAppPaaSCosts(List<WebApp_PaaS> WebApp_PaaS_List)
+            Dictionary<string, HashSet<string>> EnvironmentToUniqueIdMap = new Dictionary<string, HashSet<string>>();
+            EnvironmentToUniqueIdMap["Prod"] = new HashSet<string>();
+            Dictionary<string, HashSet<string>> VMServer_IaaSEnvironmentToUniqueIdMap = DetailsVMServer_IaaS(VM_IaaS_Server_Rehost_Perf_List).EnvironmentToUniqueIdMap;
+            Dictionary<string, HashSet<string>> SQL_IaaSEnvironmentToUniqueIdMap = DetailsSQL_IaaS(SQL_IaaS_Server_Rehost_Perf_List, SQL_IaaS_Instance_Rehost_Perf).EnvironmentToUniqueIdMap;
+            Dictionary<string, HashSet<string>> Webapp_IaaSEnvironmentToUniqueIdMap = DetailsWebapp_IaaS(WebApp_IaaS_Server_Rehost_Perf_List).EnvironmentToUniqueIdMap;
+            Dictionary<string, int> VMServerEnvironmentCountDictionary = DetailsVMServer_IaaS(VM_IaaS_Server_Rehost_Perf_List).EnvironmentCountDictionary;
+            Dictionary<string, int> SQL_IaaSEnvironmentCountDictionary = DetailsSQL_IaaS(SQL_IaaS_Server_Rehost_Perf_List, SQL_IaaS_Instance_Rehost_Perf).EnvironmentCountDictionary;
+            Dictionary<string, int> Webapp_IaaSEnvironmentCountDictionary = DetailsWebapp_IaaS(WebApp_IaaS_Server_Rehost_Perf_List).EnvironmentCountDictionary;
+
+            foreach (var kvp in VMServer_IaaSEnvironmentToUniqueIdMap)
+            {
+                if (kvp.Key.Equals("Prod"))
+                {
+                    foreach ( var item in VMServer_IaaSEnvironmentToUniqueIdMap["Prod"])
+                    {
+                        if (!EnvironmentToUniqueIdMap["Prod"].Contains(item))
+                            EnvironmentToUniqueIdMap["Prod"].Add(item);
+                    }
+                }
+            }
+
+            foreach (var kvp in SQL_IaaSEnvironmentToUniqueIdMap)
+            {
+                if (kvp.Key.Equals("Prod"))
+                {
+                    foreach (var item in SQL_IaaSEnvironmentToUniqueIdMap["Prod"])
+                    {
+                        if (!EnvironmentToUniqueIdMap["Prod"].Contains(item))
+                            EnvironmentToUniqueIdMap["Prod"].Add(item);
+                    }
+                }
+            }
+
+            foreach (var kvp in Webapp_IaaSEnvironmentToUniqueIdMap)
+            {
+                if (kvp.Key.Equals("Prod"))
+                {
+                    foreach (var item in Webapp_IaaSEnvironmentToUniqueIdMap["Prod"])
+                    {
+                        if (!EnvironmentToUniqueIdMap["Prod"].Contains(item))
+                            EnvironmentToUniqueIdMap["Prod"].Add(item);
+                    }
+                }
+            }
+
+            ManagementObj.SourceCount = EnvironmentToUniqueIdMap.Count;
+            ManagementObj.TargetCount = VMServerEnvironmentCountDictionary["Prod"] + SQL_IaaSEnvironmentCountDictionary["Prod"] + Webapp_IaaSEnvironmentCountDictionary["Prod"];
+            return ManagementObj;
+        }

```

Comment:
P1: If this logic can be moved to AzureIaaSCalculator, as PaaS won't have backup and recovery costs, we can make a lot of efficiency by eliminating the loops.